gnosis/eth/ethereum_client.py

```diff
@@ -38,7 +38,7 @@
     TimeExhausted,
     TransactionNotFound,
 )
-from web3.middleware import geth_poa_middleware
+from web3.middleware import geth_poa_middleware, simple_cache_middleware
```

Comment:
Interesting that they provide this.

It's not clear to me the default behaviour of this middleware though. They do mention in the docs that caching will apply to any call present in `rpc_whitelist`. Do we need to build this `rpc_whitelist`?

https://web3py.readthedocs.io/en/stable/middleware.html#web3.middleware.construct_simple_cache_middleware

Reply:
No, there's a default list of methods to be cached that shouldn't change: https://github.com/ethereum/web3.py/blob/master/web3/middleware/cache.py#L35

Reply:
Ah yes. I see the default argument now! Thanks.